File: хэш.cpp
Хэш-таблицы на C++:

int main() { 
// Создать хэш-таблицу 
unordered_map<string, int> umap; 
// Вставить пары «ключ-значение» 
umap["Apple"] = 10; 
umap["Mango"] = 20; 
umap["Cherry"] = 30; 
// Распечатать пары «ключ-значение» 
for (auto it : umap) cout << it.first << " " << it.second << endl; 
return 0; 
} 

#include <iostream> 
#include <list> 
#include <functional> 
// Структура для хранения пары ключ-значение 
template<typename K, typename V> 
struct HashNode { 
K key; 
V value; 
HashNode(K k, V v) : key(k), value(v) {} 
}; 
// Класс хеш-таблицы 
template<typename K, typename V> 
class HashTable { 
private: 
std::list<HashNode<K, V>>* table; 
size_t capacity; 
size_t size; 
// Простая хеш-функция 
size_t hashFunction(const K& key) { 
return std::hash<K>()(key) % capacity; 
}

File: хэш.py
Хэш-таблицы на Python:

 
class HashTable: 
    def __init__(self, size): 
        self.size = size 
        self.table = [None]*size 
 
    def _hash(self, key): 
        return ord(key[0]) % self.size 
 
class HashTable: 
    def __init__(self, size): 
        self.size = size 
        self.table = [None]*size 
 
    def _hash(self, key): 
        return ord(key[0]) % self.size 
 
    def set(self, key, value): 
        hash_index = self._hash(key) 
        self.table[hash_index] = (key, value) 
 
    def get(self, key): 
        hash_index = self._hash(key) 
        if self.table[hash_index] is not None: 
            return self.table[hash_index][1] 
 
        raise KeyError(f'Key {key} not found') 
 
    def remove(self, key): 
        hash_index = self._hash(key) 
        if self.table[hash_index] is not None: 
            self.table[hash_index] = None 
        else: 
            raise KeyError(f'Key {key} not found')